internal/backuprestore/backup.go

```diff
@@ -258,24 +259,37 @@ func (h *BRHandler) ExportOadpConfigurationToDir(ctx context.Context, toDir, oad
 	if len(dpaList.Items) == 0 {
 		return nil
 	}
+	if len(dpaList.Items) != 1 {
+		h.Log.Info("Warning: more than one DataProtectionApplication CRs found, only the reconciled one will be exported.")
+	}
 
-	// Create the directory for DPAs
+	// Create the directory for DPA
 	if err := os.MkdirAll(filepath.Join(toDir, oadpDpaPath), 0o700); err != nil {
 		return err
 	}
 
 	var secrets []string
-	// Write DPAs to a given directory
+	// Note: One DPA represents one velero instance.
+	// For the same OADP namespace, only one DPA is expected to be created,
+	// which means that there will be only one successfully reconciled DPA
+	// even if there are more than one DPA created. We only want to write
+	// the reconciled DPA to a given directory if there are multiple found.
 	for _, dpa := range dpaList.Items {
 		// Unset uid and resource version
 		dpa.SetUID("")
 		dpa.SetResourceVersion("")
 
+		ok := isDPAReconciled(&dpa)
+		if !ok {
+			h.Log.Info("The DataProtectionApplication CR is not reconciled, skipping", "name", dpa.GetName())
```

Comment:
Should this be part of the health check?

Reply:
This is pre-pivot phase. Trying to dump the live DPA to new stateroot

Reply:
But as we talked, there will be some OADP checks in advance to make sure they are set.

Reply:
Yes, that's what I mean. Shouldn't we block the upgrade if there's a DPA that hasn't reconciled?

Reply:
This check here is meant to dump the Reconciled one only if there are multiple(although it's unlikely). In theory, only one DPA should be created in one oadp namespace. But OADP doesn't reject the creation of the second one, instead it just fails it (Reconciled:False).

Reply:
A scenario in which it seems reasonable to me to block the upgrade... An unexpected and unsupported configuration. As opposed to skipping past what presumably could be the desired and unreconciled DPA? ie. what if the DPA we need is truly not reconciled at this moment?

Reply:
> Shouldn't we block the upgrade if there's a DPA that hasn't reconciled?
Yes, we should block the upgrade if there's no DPA gets reconciled.  But if there is a reconciled DPA and another non-reconciled DPA, we should continue

Reply:
> As opposed to skipping past what presumably could be the desired and unreconciled DPA? 

This may be possible. Yeah, we can discuss this further. I thought as OADP doesn't block the creation of the second one, we should not block as long as there's a working one.

Reply:
We only expect one, so more we should treat as an error.

Reply:
More validation enhancement before upgrade will be done in https://issues.redhat.com/browse/CNF-10656